Sources/Auth/Model/AuthLoggable.swift

```diff
@@ -0,0 +1,94 @@
+import Logging
+import Foundation
+
+enum AuthLoggable {
+	// swiftlint:disable identifier_name
+	case initializeDeviceLoginNetworkError(error: Error)
+	case finalizeLoginNetworkError(error: Error)
+	case finalizeDeviceLoginNetworkError(error: Error)
+	case finalizeDevicePollingLimitReached
+	case getCredentialsUpgradeTokenNetworkError(error: Error)
+	case getCredentialsScopeIsNotGranted
+	case getCredentialsClientUniqueKeyIsDifferent
+	case getCredentialsUpgradeTokenNoTokenInResponse
+	case getCredentialsRefreshTokenNetworkError(error: Error)
+	case getCredentialsRefreshTokenWithClientCredentialsNetworkError(error: Error)
+	case authLogout(reason: String, error: Error? = nil)
+	case getCredentialsRefreshTokenIsNotAvailable
+	// swiftlint:enable identifier_name
+}
+
+// MARK: - Logging
+extension AuthLoggable {
+	private static let metadataErrorKey = "error"
+	private static let metadataReasonKey = "reason"
+	
+	var loggingMessage: Logger.Message {
```

Comment:
I suspect, it could be simplified, as all the cases are just capitalized case descriptions? 🤔

Reply:
let's think about it later :) just using `"\(self)"` duplicates the metadata and produces very long messages...